lib/server.ml

```diff
@@ -107,7 +107,7 @@ let connect channel ?offer ?(no_tls=false) () =
       end
     in loop ()
   in
-  let rec negotiate_tls () =
+  let negotiate_tls make_tls_channel = let rec negotiate_tls () =
```

Comment:
I've made a commit to split that (and therefore indent the following block one level).
Same for `let generic_loop chan = let rec loop () =`